R/agg_scale.R

```diff
@@ -0,0 +1,879 @@
+#' @title Aggregate/Scale a detailed level of a variable to an aggregate level
+#'
+#' @description Aggregate counts or probabilities from a detailed level of a
+#' variable to an aggregate level or scale the detailed level values so that the
+#' detailed level aggregated together equals the aggregate level.
+#'
+#' @param dt \[`data.table()`\]\cr
+#'   Data to be aggregated or scaled.
+#' @param id_cols \[`character()`\]\cr
+#'   ID columns that uniquely identify each row of `dt`.
+#' @param value_cols \[`character()`\]\cr
+#'   Value columns that should be aggregated.
+#' @param col_type \[`character(1)`\]\cr
+#'   The type of variable that is being aggregated or scaled over. Can be either
+#'   'categorical' or 'interval'.
+#' @param col_stem \[`character(1)`\]\cr
+#'   The name of the variable to be aggregated or scaled over. If aggregating an
+#'   'interval' variable should not include the '_start' or '_end' suffix.
+#' @param mapping \[`data.table()`\]\cr
+#'   For 'categorical' variables, defines how different levels of the variable
+#'   relate to each other. For aggregating 'interval' variables, it is used to
+#'   specify intervals to aggregate to, while when scaling the mapping is
+#'   inferred from the available intervals in `dt`.
+#' @param agg_function \[`function()`\]\cr
+#'   Function to use when aggregating, can be either `sum` (for counts) or
+#'   `prod` (for probabilities).
```

Comment:
Should we add a note about aggregating/scaling rates?

Reply:
I added this to the details section
```
#' The `agg` and `scale` functions currently only work when combining counts or
#' probabilities. If the data is in rate-space then you need to convert to count
#' space first, aggregate/scale, and then convert back.
```